internal/engine/executor.go

```diff
@@ -199,6 +200,8 @@ func (m *mutantExecutor) runTests(pkg string) mutator.Status {
 
 	cmd := m.execContext(ctx, "go", m.getTestArgs(pkg)...)
 	cmd.Dir = m.mutant.Workdir()
+	cmd.Env = append(cmd.Env, os.Environ()...)
```

Comment:
Wouldn't be better to do `cmd.Env = os.Environ`. Not sure, but I think we can spare an array allocation. 

At the beginning Env should be nil, in the first append, it allocates an array with the length of Environ. Then, in the second array it reallocates for the added element, isn't it?
If we assign Environ directly, we are sure at most it reallocates in the only append we have. I'm not certain my reasoning is correct though :smile:. What do you thin?

Reply:
The only reason I did that is to not break test in: `TestMutatorTestExecution`. In the `executor_test.go:557` we set the `GO_TEST_PROCESS=1` if we do not append we will override that breaking 2 tests.

About the allocation, since the os.Environ should be more or less the same every time we call it (at least the env vars we need). We might want to refactor where we assign that line.
We can create our custom execContext func that pre-allocates the env vars, so we will do the allocation and the sys call only once, and not every time we call the `runTests` func:

```go
func NewExecutorDealer(mod gomodule.GoModule, wdd workdir.Dealer, elapsed time.Duration, opts ...ExecutorDealerOption) *MutantExecutorDealer {
...
	jd := MutantExecutorDealer{
                ...
		execContext:       commandWithEnvs,
	}
...
}
func commandWithEnvs(ctx context.Context, name string, arg ...string) *exec.Cmd {
	cmd := exec.CommandContext(ctx, name, arg...)
	cmd.Env = os.Environ()
	return cmd
}

```

Reply:
Even if the env is set at every mutator run (potentially a lot of times) I was concerned, but maybe what we gain not reallocating is marginal and doesn't justify this refactoring, what do you think?

Reply:
We should perform some benchmarks. I think it won't be the most expensive operation we're doing so we should be fine for now running that code for every mutator.